helm-framework/helm/provider.go

```diff
@@ -339,10 +338,9 @@ func (p *HelmProvider) Configure(ctx context.Context, req provider.ConfigureRequ
 	if resp.Diagnostics.HasError() {
 		return
 	}
-	// Override environment variables if the configuration values are provided
-	if !config.Debug.IsNull() {
-		debug = fmt.Sprintf("%t", config.Debug.ValueBool())
-	}
+
+	debug := os.Getenv("HELM_DEBUG") == "true" || config.Debug.ValueBool()
```

Comment:
nice one liner! One last request

Here `debug` is initialized on line 342 but is first used here in the `provider.go` file:
https://github.com/hashicorp/terraform-provider-helm/blob/2eb62ef550dbb1858ef03248b93fc6f53e98e36d/helm-framework/helm/provider.go#L460

It's best practice to include the variable initialization near the first call of the variable. In order to reduce the need to go searching for where the variable was first set.